File: backend/app/repositorios/ImagenSQLiteRepo.go
package repositorios

import(
	"database/sql"
	"mizcuin/modelos"
	"os"
	"fmt"
	_ "github.com/mattn/go-sqlite3"
)

var dbPathImagen string

func init(){
	dbPathImagen = os.Getenv("DB_PATH")
}

func GuardarImagen(consulta string) bool{
	db, err := sql.Open("sqlite3", dbPathImagen)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en imagen", err)
		return false
	}

	defer db.Close()

	_, err = db.Exec(consulta)
	if err != nil {
		fmt.Println("Error al ejecutar la consulta en imagen", err)
		return false
	}

	fmt.Println("Imagen guardada correctamente: ", consulta)
	return true
	
}

func BuscarImagenes(consulta string) (modelos.ListadoImagenes, bool){
	db, err := sql.Open("sqlite3", dbPathImagen)

	if err != nil {
		fmt.Println("Error al abrir la base de datos en imagen", err)
		return modelos.ListadoImagenes{}, false
	}

	defer db.Close()

	rows, err := db.Query(consulta)

	if err != nil {
		fmt.Println("Error al buscar la imagen", err)
		return modelos.ListadoImagenes{}, false
	}

	defer rows.Close()

	var imagenes []modelos.Imagen

	for rows.Next() {
		var idImagen int
		var idTrabajo int
		var imagen []byte

		err = rows.Scan(&idImagen, &idTrabajo, &imagen)

		if err != nil {
			fmt.Println("Error al buscar la imagen inesperado ", err)
			return modelos.ListadoImagenes{}, false
		}

		imagenes = append(imagenes, modelos.Imagen{Id_trabajo: idTrabajo, Imagen: imagen})
	}

	fmt.Println("Imagenes encontradas correctamente: ", consulta)
	return modelos.ListadoImagenes{Imagenes: imagenes}, true
}

func ActualizarImagen(consulta string) bool{
	return GuardarImagen(consulta)
}

func EliminarImagen(consulta string) bool{
	return GuardarImagen(consulta)
}

File: backend_Python/app/repositorios/imagenSQLiterepo.py
import os
import sqlite3

from modelos.Imagen import Imagen
from dto.listadoImagenes import ListadoImagenes

DB_PATH = os.getenv("DB_PATH")

def guardar_imagen(consulta):
    try:
        conn = sqlite3.connect(DB_PATH)
        cursor = conn.cursor()

        cursor.execute(consulta)
        conn.commit()
        conn.close()

        print("Imagen guardada correctamente:", consulta)
        return True

    except Exception as e:
        print("Error al ejecutar la consulta en imagen:", e)
        return False
    
def buscar_imagenes(consulta):
    try:
        conn = sqlite3.connect(DB_PATH)
        cursor = conn.cursor()

        cursor.execute(consulta)
        filas = cursor.fetchall()
        conn.close()

        imagenes = []

        for fila in filas:
            # fila = (id_imagen, id_trabajo, imagen_blob)
            imagen = Imagen(
                fila[0],  # id_imagen
                fila[1],  # id_trabajo
                fila[2]   # imagen (bytes)
            )
            imagenes.append(imagen)

        print("Imagenes encontradas correctamente:", consulta)
        return ListadoImagenes(imagenes), True

    except Exception as e:
        print("Error al buscar las imagenes:", e)
        return ListadoImagenes([]), False
    
def actualizar_imagen(consulta):
    return guardar_imagen(consulta)

def eliminar_imagen(consulta):
    return guardar_imagen(consulta)